aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    assert_success,
+    tests::{aggregator_v2::EAGGREGATOR_OVERFLOW, common},
+    MoveHarness,
+};
+use aptos_language_e2e_tests::account::Account;
+use aptos_types::{
+    on_chain_config::FeatureFlag,
+    transaction::{ExecutionStatus, SignedTransaction, TransactionOutput},
+};
+use claims::{assert_matches, assert_ok, assert_some};
+use move_core_types::{
+    account_address::AccountAddress, ident_str, language_storage::ModuleId,
+    parser::parse_struct_tag, vm_status::AbortLocation,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize)]
+struct Aggregator {
```

Comment:
Why are these definitions required here? We didn't have to define these for other aggregator e2e tests so far right?

Reply:
So to test things and read a resource there are a few options:
1. What we did with aggregators tests, that is have a `check` transaction which does asserts.
2. What I did here: read a resource from state view and deserialize it into a rust struct (essentially there is a 1:1 mapping between Move values and this). Annotations allow to use bcs for serialization/deserialization.
3. Use view functions (haven't tried it before myself), but you can in theory define a view function which can give you certain values of a resource.